itests/kit/node_full.go

```diff
@@ -180,6 +181,139 @@ func (f *TestFullNode) ExpectSend(ctx context.Context, from, to address.Address,
 	return nil
 }
 
+func (f *TestFullNode) IsSyncedWith(ctx context.Context, from abi.ChainEpoch, nodes ...*TestFullNode) (abi.ChainEpoch, error) {
+	if len(nodes) < 1 {
+		return 0, fmt.Errorf("no checked nodes")
+	}
+	base, err := ChainHeadWithCtx(ctx, f)
+	if err != nil {
+		return 0, err
+	}
+
+	to := base.Height()
+	for h := from; h <= to; h++ {
+		h := h
+
+		baseTipSet, err := f.ChainGetTipSetByHeight(ctx, h, types.EmptyTSK)
+		if err != nil {
+			return 0, err
+		}
+		if baseTipSet.Height() != h {
+			return 0, fmt.Errorf("couldn't find tipset for height %d in base node", h)
+		}
+
+		// TODO: We can probably parallelize the check for each node?
+
+		g, ctx := errgroup.WithContext(ctx)
+
+		for _, n := range nodes {
+			n := n
+
+			// We don't need to check that base node is in sync with itself.
+			if n == f {
+				continue
+			}
+			g.Go(func() error {
+				return n.waitForTipSet(ctx, h, baseTipSet)
+			})
+		}
+
+		if err := g.Wait(); err != nil {
+			return 0, err
+		}
+	}
+
+	// Additional check of the invariant.
+	baseTipSet, err := f.ChainGetTipSetByHeight(ctx, to, types.EmptyTSK)
+	if err != nil {
+		return 0, err
+	}
+
+	for _, n := range nodes {
+		ts, err := n.ChainGetTipSetByHeight(ctx, to, types.EmptyTSK)
+		if err != nil {
+			return 0, err
+		}
```

Comment:
This may be a source of flakiness. `types.EmptyTSK` gets the heaviest tipset but you may be comparing at different heights as nodes may have moved on, you need to check against the specific tipset height.
```suggestion
		ts, err := n.ChainGetTipSetByHeight(ctx, to, baseTipSet.Height())
		if err != nil {
			return 0, err
		}
```